pkg/kvstore/consul.go

```diff
@@ -500,15 +500,20 @@ func (c *consulClient) UpdateIfDifferent(ctx context.Context, key string, value
 	getR, _, err := c.KV().Get(key, nil)
 	increaseMetric(key, metricRead, "Get", duration.EndError(err).Total(), err)
 	// On error, attempt update blindly
-	if err == nil && getR != nil {
-		// if lease is different and value is not equal then update.
-		if getR.Session == c.lease &&
-			bytes.Equal(getR.Value, value) {
-			return false, nil
-		}
+	if err != nil || getR == nil {
```

Comment:
Not sure if this would be helpful - would add debug / trace logs indicating the case that is triggering the update of the kvstore given that the outcome is the same for this case, and the one on 507 and 512 as well provide any insight for debugging in the future?